### ISSUE TITLE
`@stylistic/no-extra-parens`: False-positive on TSDoc type-casts

### ISSUE THREAD

**Issue by SamuelT-Beslogic** [eslint-stylistic/eslint-stylistic#1088]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Parentheses are necessary to do type-casting using TSDoc comments in pure JS files.

```js
export const jestNoRestrictedSyntax = (
  /** @type {readonly (number | string | {readonly selector: string})[]} */ noRestrictedSyntax
) => /** @type {import('@eslint/config-helpers').RuleConfig} */ (noRestrictedSyntax.filter(
  entry =>
    typeof entry === "number"
    || typeof entry === "string"
    // Exclude from jest tests for compatibility with
    // `jest/valid-describe-callback` and `jest/no-test-return-statement`
    || !entry.selector.startsWith("ArrowFunctionExpression")
))
```

### Reproduction

https://stackblitz.com/edit/github-bvwruhsz?file=src%2Findex.js

### System Info

```shell
System:
    OS: Windows 11 10.0.26200
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
    Memory: 14.65 GB / 31.73 GB
  Binaries:
    Node: 20.19.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.22 - C:\Program Files\nodejs\yarn.CMD
    npm: 11.6.2 - C:\Program Files\nodejs\npm.CMD
    pnpm: 10.23.0 - C:\Program Files\nodejs\pnpm.CMD
  npmPackages:
    @stylistic/eslint-plugin: ^5.3 => 5.6.1
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
I think [`allowParensAfterCommentPattern`](https://eslint.style/rules/no-extra-parens#allowparensaftercommentpattern) should help resolve this.

**Comment by SamuelT-Beslogic** [comment]
Thanks for the ref! I'll configure it for now in my shared settings to avoid false-positives. But I feel like this use-case should probably work out of the box since it's not a stylistic choice. imo even just having a default config for `allowParensAfterCommentPattern`, that is also documented in cast the user overwrites it, should be enough.

Edit: just to clarify. The current doc does use type casting as an example for that exact purpose. By "documented" I mean that if the default value changes, said default config should be documented. The pattern could probably be a tiny bit more restricted by using word boundaries: `"\\b@type\\b"` <-- Doesn't pickup `@types`
